src/net.h

```diff
@@ -1267,6 +1267,36 @@ class CConnman
      */
     std::vector<CService> m_onion_binds;
 
+    /**
+     * RAII helper to atomically create a copy of `vNodes` and add a reference
+     * to each of the nodes. The nodes are released when this object is destroyed.
+     */
+    class NodesSnapshot
+    {
+    public:
+        explicit NodesSnapshot(CConnman* connman) : m_connman{connman}
+        {
+            LOCK(m_connman->cs_vNodes);
+            m_nodes_copy = m_connman->vNodes;
+            for (auto& node : m_nodes_copy) {
+                node->AddRef();
+            }
+        }
+
+        ~NodesSnapshot()
+        {
+            LOCK(m_connman->cs_vNodes);
+            for (auto& node : m_nodes_copy) {
+                node->Release();
+            }
+        }
+
+        std::vector<CNode*> m_nodes_copy;
```

Comment:
96bcc72b5dc4f8ba5654d8ddeab844688507f1ce

make private and add const getter `const std::vector<CNode*>& nodes() const { return m_nodes_copy; }`

Reply:
Done.